File: Python_Scripting_for_Computational_Science_Third_Edition/py/gui/hwGUI7.py
#!/usr/bin/env python
from Tkinter import *
import math

root = Tk()              # root (main) window
top = Frame(root)        # create frame
top.pack(side='top')     # pack frame in main window

# create frame to hold the first widget row:
hwframe = Frame(top)
# this frame (row) is packed from top to bottom (in the top frame):
hwframe.pack(side='top')
# create label in the frame:
font = 'times 18 bold'
hwtext = Label(hwframe, text='Hello, World!', font=font)
hwtext.pack(side='top', pady=20)

# create frame to hold the middle row of widgets:
rframe = Frame(top)
# this frame (row) is packed from top to bottom:
rframe.pack(side='top', padx=10, pady=20)
# create label and entry in the frame and pack from left:
r_label = Label(rframe, text='The sine of')
r_label.pack(side='left')
r = StringVar() # variable to be attached to r_entry
r.set('1.2')    # default value
r_entry = Entry(rframe, width=6, textvariable=r)
r_entry.pack(side='left')

s = StringVar() # variable to be attached to s_label
def comp_s(event):
    global s
    s.set('%g' % math.sin(float(r.get()))) # construct string

r_entry.bind('<Return>', comp_s)

compute = Label(rframe, text=' equals ')
compute.pack(side='left')

s_label = Label(rframe, textvariable=s, width=12)
s_label.pack(side='left')

# finally, make a quit button and a binding of q to quit:

def quit(event=None):
    root.destroy()

quit_button = Button(top, text='Goodbye, GUI World!', command=quit,
                     background='yellow', foreground='blue')
quit_button.pack(side='top', pady=5, fill='x')

root.bind('<q>', quit)

root.mainloop()

File: Python_Scripting_for_Computational_Science_Third_Edition/perl/hwGUI7.pl
: # *-*-perl-*-*
  eval 'exec perl -w -S  $0 ${1+"$@"}' 
    if 0;  # if running under some shell

use Tk;
$main_win = MainWindow->new();
$top = $main_win->Frame(); 
$top->pack(-side => 'top');

# create a frame to hold the first "Hello, World!" widget:
$hwframe = $top->Frame();
# the row frames are packed from top to bottom:
$hwframe->pack(-side => 'top');
# create a label inside this frame:
$font = "times 18 bold";
$hwtext = $hwframe->Label(-text => "Hello, World!", -font => $font);
# pack the label inside the frame:
$hwtext->pack(-side => 'left', -pady => 20);  
# (side does not matter)

# create a frame to hold "The sine of [number] equals [result]":
$rframe = $top->Frame();
# pack row frames from top to bottom:
$rframe->pack(-side => 'top', , -pady => 20, -padx => 10);
# create labels and textentry, packed from the left, as childred
# of the $rframe frame:
$rframe->Label(-text => 'The sine of ')->pack(-side => 'left');
						
$r = 1.2;  # default
$r_entry = $rframe->Entry(-width => 6, -relief => 'sunken',
                          -textvariable => \$r);
$r_entry->pack(-side => 'left');
$r_entry->bind('<Return>', \&comp_s);
sub comp_s { $s = sin($r); }
$rframe->Label(-text => " equals ")->pack(-side => 'left');
$rframe->Label(-textvariable => \$s, 
               -width => 18)->pack(-side => 'left');

# create a quit button, but drop the frame as this is only one widget:
$top->Button(-text => 'Goodbye, GUI World!', -command => \&exit,
             -background => 'yellow',  -foreground => 'blue')
             ->pack(-side => 'top', -pady => 5, -fill => 'x');

$main_win->bind('<q>', \&exit);

MainLoop();





